pandas/core/groupby/ops.py

```diff
@@ -608,9 +618,52 @@ def _ea_wrap_cython_operation(
             f"function is not implemented for this dtype: {values.dtype}"
         )
 
+    @final
+    def _masked_ea_wrap_cython_operation(
+        self,
+        kind: str,
+        values: BaseMaskedArray,
+        how: str,
+        axis: int,
+        min_count: int = -1,
+        **kwargs,
+    ) -> BaseMaskedArray:
+        """
+        Equivalent of `_ea_wrap_cython_operation`, but optimized for masked EA's
+        and cython algorithms which accept a mask.
+        """
+        orig_values = values
+
+        # isna just directly returns self._mask, so copy here to prevent
+        # modifying the original
+        mask = isna(values).copy()
```

Comment:
Values in `mask`  are only accessed, and `mask` is never modified, I think? So the copy should not be needed? (otherwise the same argument would hold for the `arr = values._data` on the line below?)

I would maybe just do `mask = values._mask` for consistency with `values._data` (and because the fact that `isna` doesn't return a copy seems like a bug to me)

Reply:
Yeah the copy is not needed here. I had the copy originally because I was thinking more generally, eg for some algos like `cumsum`, NaN's can be added in places they originally weren't present, like

```
>>> ser = pd.Series([np.nan, 1, 2])
>>> ser.groupby([1, 1, 1]).cumsum(skipna=False)
0   NaN
1   NaN
2   NaN
``` 


So modifying the input mask in place would be a good way to support this for nullable types. But agreed may as well not copy until necessary. 

`values._mask` point makes sense, that `isna` behavior is surprising.

Reply:
Actually, I think we do need to copy `mask`, because otherwise the input array and result array will have masks sharing the same memory. `arr` doesn't need to be copied because the algorithm modifies an `out` array built from scratch, so no memory sharing issues there

Reply:
> Actually, I think we do need to copy mask, because otherwise the input array and result array

Ah, yes, good catch. And thanks for adding a comment about that.